### ISSUE TITLE
Differences between FrameNet in Python3.6 and Python2.7

### ISSUE THREAD

**Issue by alvations** [nltk/nltk#1901]
Framenet in Python3.6 and Python2.7 seems to behave differently:

From https://github.com/nltk/nltk/blob/develop/nltk/test/framenet.doctest#L119

**Python3.6**

```python
~/git-stuff/nltk$ python3
Python 3.6.1 (default, Apr  4 2017, 09:40:21) 
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from nltk.corpus import framenet as fn
>>> fn.frames_by_lemma(r'(?i)a little') 
[<frame ID=2001 name=Degree>, <frame ID=189 name=Quantified_mass>]
>>> exit()
```

**Python2.7**

```python
~/git-stuff/nltk$ python
Python 2.7.13 (default, Dec 18 2016, 07:03:39) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from nltk.corpus import framenet as fn
>>> fn.frames_by_lemma(r'(?i)a little')
[<frame ID=189 name=Quantified_mass>, <frame ID=2001 name=Degree>]
```

Is the order of the frames important?

@nschneid any hints on why this is happening?

----

Another instance is during the doctest at https://github.com/nltk/nltk/blob/develop/nltk/corpus/reader/framenet.py#L2250

**Python 3.6**

```
~/git-stuff/nltk$ python3
Python 3.6.1 (default, Apr  4 2017, 09:40:21) 
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from nltk.corpus import framenet as fn
>>> from nltk.corpus.reader.framenet import PrettyDict
>>> frts = list(fn.frame_relation_types())
>>> PrettyDict(frts[0], breakLines=True)
{'ID': 10,
 '_type': 'framerelationtype',
 'frameRelations': [<Causative=Cause_to_move_in_place -- Causative_of -> Inchoative/state=Moving_in_place>, <Causative=Apply_heat -- Causative_of -> Inchoative/state=Absorb_heat>, ...],
 'name': 'Causative_of',
 'subFrameName': 'Inchoative/state',
 'superFrameName': 'Causative'}
```

**Python 2.7**

```
~/git-stuff/nltk$ python2
Python 2.7.13 (default, Dec 18 2016, 07:03:39) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from nltk.corpus import framenet as fn
>>> from nltk.corpus.reader.framenet import PrettyDict
>>> frts = list(fn.frame_relation_types())
>>> PrettyDict(frts[0], breakLines=True)
{'ID': 1,
 u'_type': u'framerelationtype',
 u'frameRelations': [<Parent=Event -- Inheritance -> Child=Change_of_consistency>, <Parent=Event -- Inheritance -> Child=Rotting>, ...],
 'name': 'Inheritance',
 'subFrameName': 'Child',
 'superFrameName': 'Parent'}
>>> exit()
```

----

Another instance is at https://github.com/nltk/nltk/blob/develop/nltk/corpus/reader/framenet.py#L1914

**Python 3.6**

```
~/git-stuff/nltk$ python3
Python 3.6.1 (default, Apr  4 2017, 09:40:21) 
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from nltk.corpus import framenet as fn
>>> from nltk.corpus.reader.framenet import PrettyList
>>> PrettyList(fn.lus(r'(?i)a little'), maxReprSize=0, breakLines=True)
[<lu ID=14744 name=a little bit.adv>,
 <lu ID=14743 name=a little.adv>,
 <lu ID=14733 name=a little.n>]
```

**Python 2.7**

```
>>> from nltk.corpus import framenet as fn
>>> from nltk.corpus.reader.framenet import PrettyList
>>> PrettyList(fn.lus(r'(?i)a little'), maxReprSize=0, breakLines=True)
[<lu ID=14744 name=a little bit.adv>,
 <lu ID=14733 name=a little.n>,
 <lu ID=14743 name=a little.adv>]
```

Are orders in the frames and lists important?

**Comment by nschneid** [comment]
The lists are constructed by iterating over a dict (returned by `frame_ids_and_names()`). The order is not important. These methods might as well return sets, though I think for compatibility with Python <2.7 I was avoiding set literals and comprehensions.

**Comment by nschneid** [comment]
Oh I see, these return `PrettyList` instances for pretty-printing, and there is no `PrettySet`.

**Comment by alvations** [comment]
Thanks @nschneid! I'll put all of them under `set()` and check for sequence equality =)

**Comment by alvations** [comment]
@nschneid how about `frame_relations()` and `fe_relations` is order unimportant for those too? https://github.com/nltk/nltk/blob/develop/nltk/corpus/reader/framenet.py#L2277 

I think there's some inherent sorting at https://github.com/nltk/nltk/blob/develop/nltk/corpus/reader/framenet.py#L2345 and https://github.com/nltk/nltk/blob/develop/nltk/corpus/reader/framenet.py#L2377

**Comment by nschneid** [comment]
Right, there is purposeful sorting in `frame_relations()` and `fe_relations()`, so those should be left alone.

**Comment by alvations** [comment]
Resolved in #1902
